xla/service/unique_channel_id_enforcer.cc

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2024 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "xla/service/unique_channel_id_enforcer.h"
+
+#include "xla/hlo/ir/hlo_module.h"
+#include "xla/hlo/utils/hlo_query.h"
+#include "absl/status/statusor.h"
+
+namespace xla {
+
+absl::StatusOr<bool> UniqueChannelIdEnforcer::Run(
+    HloModule* module,
+    const absl::flat_hash_set<absl::string_view>& execution_threads) {
+  absl::flat_hash_set<std::optional<long int>> used_channel_ids;
+  auto next_channel_id = hlo_query::NextChannelId(*module);
+  bool changed = false;
+  for (HloComputation* computation : module->computations()) {
+    for (HloInstruction* instruction : computation->instructions()) {
+      if (hlo_query::IsCollectiveCommunicationOp(instruction->opcode())) {
```

Comment:
In general we prefer early-return type programming. The justification is to reduce nesting. 

So negating the condition + continue is preferred.

Reply:
Done